example/service/s3/getObjectWithProgress/getObjectWithProgress.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"regexp"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"strings"
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+type progressWriter struct {
```

Comment:
Could you add documentation to the `progressWriter` type about how it works, and how the progress is printed?

Reply:
Done! 👍